### PR TITLE
Migrate existing profile names from Person X to Profile X

### PR DESCRIPTION
**Pull request by petemill**
Chromium does [this type of thing in the constructor of ProfileInfoCache](https://cs.chromium.org/chromium/src/chrome/browser/profiles/profile_info_cache.cc?g=0&l=714). The constructor of BraveProfileManager seems just as appropriate. Example contained in the BrowserTest I added.

Fix https://github.com/brave/brave-browser/issues/5814

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See https://github.com/brave/brave-browser/issues/5814 for detailed test plan

### From this
image

### To this
image

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/profiles/brave_profile_manager_browsertest.cc [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  const std::vector<TestProfileData> profile_data = {
+    {
+      base::ASCIIToUTF16("Person 1"),
+      base::ASCIIToUTF16("Profile 1"), true,
+        profile_manager->user_data_dir().Append(
+          profile_manager->GetInitialProfileDir())},
+    {
+      base::ASCIIToUTF16("Person 2"),
+      base::ASCIIToUTF16("Profile 2"), true,
+        profile_manager->user_data_dir().Append("testprofile2")},
```
Comment (simonhong) [inline browser/profiles/brave_profile_manager_browsertest.cc]: `FILE_PATH_LITERAL("testprofile2")` should be used for Windows.

File: browser/profiles/brave_profile_manager_browsertest.cc [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    {
+      base::ASCIIToUTF16("ZZCustom 3"),
+      base::ASCIIToUTF16("ZZCustom 3"), false,
+        profile_manager->user_data_dir().Append("testprofile3")},
```
Comment (simonhong) [inline browser/profiles/brave_profile_manager_browsertest.cc]: ditto

File: browser/profiles/brave_profile_manager_browsertest.cc [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  bool has_entry1 = storage.GetProfileAttributesWithPath(
+      profile_data[0].profile_path, &entry1);
+  ASSERT_EQ(has_entry1, true);
+  entry1->SetName(profile_data[0].profile_name);
+  if (profile_data[0].force_default_name) {
+    entry1->SetIsUsingDefaultName(true);
+  }
+  // Rest are generated
+  for (size_t i = 0; i != profile_data.size(); i++) {
+    /* std::cout << v[i]; ... */
```
Comment (simonhong) [inline browser/profiles/brave_profile_manager_browsertest.cc]: nit: delete

**Comment by petemill** [comment]
@simonhong all feedback addressed - thanks!

**Comment by petemill** [comment]
I just noticed a test name that mistakenly got renamed with a find / replace that I'd like to correct.

File: browser/profiles/brave_profile_manager_browsertest.cc [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  };
+  return profile_data;
+}
+
+}  // namespace
+
+// Test that legacy profile names (Person X) that have
+// not been user-modified are automatically renamed
+// to brave profile names (Profile X).
+IN_PROC_BROWSER_TEST_F(BraveProfileManagerTest, PRE_Migrateprofile_names) {
```
Comment (darkdh) [inline browser/profiles/brave_profile_manager_browsertest.cc]: we should avoid underscore as best practice https://github.com/google/googletest/blob/master/googletest/docs/faq.md#why-should-test-suite-names-and-test-names-not-contain-underscore unless we are using special prefix like `DISABLED`, `MANUAL`, `MAYBE`, `PRE`...etc. so this should be like `PRE_MigrateProfileNames`

File: browser/profiles/brave_profile_manager_browsertest.cc [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    ProfileAttributesEntry* entry;
+    bool has_entry = storage.GetProfileAttributesWithPath(
+        profile_data[i].profile_path, &entry);
+    ASSERT_EQ(has_entry, true);
+    entry->SetName(profile_data[i].profile_name);
+    entry->SetIsUsingDefaultName(profile_data[i].force_default_name);
+  }
+}
+
+IN_PROC_BROWSER_TEST_F(BraveProfileManagerTest, Migrateprofile_names) {
```
Comment (darkdh) [inline browser/profiles/brave_profile_manager_browsertest.cc]: ditto

File: browser/profiles/brave_profile_manager_browsertest.cc (hunk 4 above)
Comment (petemill) [reply browser/profiles/brave_profile_manager_browsertest.cc]: @darkdh yeah as stated above, that was due to some over-eager find/replace! That's fixed now via https://github.com/brave/brave-core/compare/a0d154f196ae541be59dd286f6d21b11006b5318..af4f1e901aba92a3e19ac4263a95368c777b7fe9

**Pull request merged**